client/web-sveltekit/src/lib/repo/open-in-editor/OpenInEditor.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+    import { type EditorSettings, getEditor, parseBrowserRepoURL, buildRepoBaseNameAndPath, buildEditorUrl } from '$lib/web'
+    import { getEditorSettingsErrorMessage } from './build-url'
+    import Tooltip from '$lib/Tooltip.svelte'
+    import EditorIcon from '$lib/repo/open-in-editor/EditorIcon.svelte'
+    import { settings } from '$lib/stores'
+    import { mdiCodeBraces } from '@mdi/js'
+    import Icon from '$lib/Icon.svelte';
+    import {SourcegraphURL} from '$lib/common';
+    import {page} from '$app/stores';
+
+    export let externalServiceType: string = ''
+
+    let openInEditor = $settings?.openInEditor
+
+    const editorSettingsErrorMessage = getEditorSettingsErrorMessage(openInEditor)
+    const editorIds = openInEditor?.editorIds ?? []
+    const editors = !editorSettingsErrorMessage ? editorIds.map(getEditor) : undefined
+
+    const sourcegraphBaseURL = SourcegraphURL.from($page.url).toString();
+
+    const { repoName, filePath, position, range } = parseBrowserRepoURL(window.location.href)
+    const start = position ?? range?.start
```

Comment:
These should all be reactive otherwise the code only runs once when the component is created. You should also use `$page.url` instead of `window.location.href`, otherwise the switching to a new URL won't update the variables.

```suggestion
    let openInEditor = $settings?.openInEditor

    $: editorSettingsErrorMessage = getEditorSettingsErrorMessage(openInEditor)
    $: editorIds = openInEditor?.editorIds ?? []
    $: editors = !editorSettingsErrorMessage ? editorIds.map(getEditor) : undefined

    $: sourcegraphBaseURL = SourcegraphURL.from($page.url).toString();

    $: { repoName, filePath, position, range } = parseBrowserRepoURL($page.url)
    $: start = position ?? range?.start
```

Reply:
Destructuring needs another pair of brackets around the statement, but apart from that everything works!

```
$: ({ repoName, filePath, position, range } = parseBrowserRepoURL($page.url.toString()))
```